File: Image_Compression/temporal_sptl2.m
function [pred11] = temporal_sptl2(D,c)
const=5;
[row,col]=size(D);
pred1=zeros(row+1,col+1);
pred11=zeros(row+1,col+1);

pred1(1,:)=const*ones(1,col+1);
pred1(2:row+1,1)=const*ones(row,1);

pred1(2:row+1,2:col+1)=D;

switch c

    case 1
        for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=pred1(i-1,j);
            end
        end
    case 2
        for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=pred1(i-1,j-1);
            end
        end
    case 3
        for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=pred1(i,j-1);
            end
        end
 %(A+B)/2       
    case 4
        for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=0.5*(pred1(i,j-1)+pred1(i-1,j));
            end
        end

%A+B-C
    case 5

        for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=pred1(i,j-1)+pred1(i-1,j)-pred1(i-1,j-1);
            end
        end
        
%A+(B-C)/2
    case 6
         for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=pred1(i,j-1)+((pred1(i-1,j)-pred1(i-1,j-1))*0.5);
            end
         end
         
%B+(A-C)/2
    case 7
          for i=2:row+1
            for j=2:col+1    
                pred11(i,j)=pred1(i-1,j)+((pred1(i,j-1)-pred1(i-1,j-1))*0.5);
            end
          end
          
 end
          
 

pred11=pred11(2:row+1,2:col+1);
pred11=D-fix(pred11);
pred11=double(pred11);

File: Image_Compression/temporal_sptl2.py
import numpy as np

def temporal_sptl2(D, c):
    const = 5
    [row, col] = np.shape(D)
    pred1 = np.zeros((row+1, col+1))
    pred11 = np.zeros((row+1, col+1))

    pred1[0, :] = const * np.ones((1, col+1))
    pred1[1: row+1, 0] = const
    pred1[1: row+1, 1:col+1] = D
    # pred1[0, :] = const
    # #print(pred1)

    # pred1[1:row, 0] = const
    # #print(pred1)

    # pred1[1:row, 1:col] = D
    # print(pred1)
    # print(pred11)

    if c==1:    #X=B
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = pred1[i-1][j]
    if c==2:    #X=C
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = pred1[i-1][j-1]
    if c==3:    #X=A
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = pred1[i][j-1]
    if c==4:    #X=(A+B)/2
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = 0.5 * (pred1[i][j-1] + pred1[i-1][j])
    if c==5:    #X=A+B-C
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = pred1[i][j-1] + pred1[i-1][j] - pred1[i-1][j-1]
    if c==6:
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = pred1[i][j-1] + (0.5 * (pred1[i-1][j] - pred1[i-1][j-1]))
    if c==7:
        for i in range(1, row):
            for j in range(1, col):
                pred11[i][j] = pred1[i-1][j] + (0.5 * (pred1[i][j-1] - pred1[i-1][j-1]))
    
    pred11 = pred11[1:row+1, 1:col+1]
    # print(pred11)
    pred11 = D - np.fix(pred11)
    # print(pred11)
    # pred11 = pred11.astype('double')
    # print(pred11)

    return pred11